### ISSUE TITLE
Wrong World File Version Error

### ISSUE THREAD

**Issue by Darkstar0001** [TEdit/Terraria-Map-Editor#829]
I'm trying to load a world that I've edited a lot in TEdit, but for some reason it is telling me that the world version isn't supported. I downloaded the latest version of TEdit, but it still didn't work. TEdit opens other worlds just fine, including the in-game back-up of the world in question (but not the TEdit backup). It's probably worth mentioning that my computer crashed the last time I was editing the world (most likely not due to TEdit, though).

![image](https://cloud.githubusercontent.com/assets/19754911/16443496/dcbb0698-3d9e-11e6-95ba-240fcbafa8f5.png)

**ETA:** I tried force-loading the world, but didn't get far. World properties and kill tallies seemed in order, but when I hit analyze, the program crashed, and told me to copy from the log. So here you go:

06-29-2016 02:04:42: Starting TEdit 3.7.16172.2152 
06-29-2016 02:04:42: OS: Microsoft Windows NT 6.1.7601 Service Pack 1 
06-29-2016 02:04:43: Terraria Data Path: c:\program files (x86)\steam\steamapps\common\terraria\Content 
06-29-2016 02:04:48: Creating Undo cache: C:\Users\AWS\AppData\Roaming\TEdit\undo_465308768 
06-29-2016 02:14:11: Removing old undo cache: C:\Users\AWS\AppData\Roaming\TEdit\undo_340549927 
06-29-2016 02:14:11: Removing old undo cache: C:\Users\AWS\AppData\Roaming\TEdit\undo_516148657 
06-29-2016 02:14:12: Removing old undo cache: C:\Users\AWS\AppData\Roaming\TEdit\undo_693119263 
06-29-2016 02:14:43: Error - Object reference not set to an instance of an object.
   at TEditXNA.Terraria.World.Validate() in c:\projects\terraria-map-editor\TEditXna\Terraria\World.cs:line 214
   at TEditXNA.Terraria.WorldAnalysis.WriteAnalyzeWorld(StreamWriter sb, World world, Boolean fullAnalysis) in c:\projects\terraria-map-editor\TEditXna\Terraria\WorldAnalysis.cs:line 51
   at TEditXNA.Terraria.WorldAnalysis.AnalyzeWorld(World world) in c:\projects\terraria-map-editor\TEditXna\Terraria\WorldAnalysis.cs:line 30
   at TEditXna.ViewModel.WorldViewModel.AnalyzeWorld() in c:\projects\terraria-map-editor\TEditXna\ViewModel\WorldViewModel.cs:line 645 
06-29-2016 02:14:43: Error - Exception has been thrown by the target of an invocation.
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at GalaSoft.MvvmLight.Helpers.WeakAction.Execute() in D:\GalaSoft\mydotnet\MVVMLight\source\GalaSoft.MvvmLight\GalaSoft.MvvmLight (PCL)\Helpers\WeakAction.cs:line 301
   at GalaSoft.MvvmLight.Command.RelayCommand.Execute(Object parameter) in D:\GalaSoft\mydotnet\MVVMLight\source\GalaSoft.MvvmLight\GalaSoft.MvvmLight (PCL)\Command\RelayCommand.cs:line 223
   at MS.Internal.Commands.CommandHelpers.CriticalExecuteCommandSource(ICommandSource commandSource, Boolean userInitiated)
   at System.Windows.Controls.Primitives.ButtonBase.OnClick()
   at System.Windows.Controls.Button.OnClick()
   at System.Windows.Controls.Primitives.ButtonBase.OnMouseLeftButtonUp(MouseButtonEventArgs e)
   at System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
   at System.Windows.RoutedEventHandlerInfo.InvokeHandler(Object target, RoutedEventArgs routedEventArgs)
   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   at System.Windows.UIElement.ReRaiseEventAs(DependencyObject sender, RoutedEventArgs args, RoutedEvent newEvent)
   at System.Windows.UIElement.OnMouseUpThunk(Object sender, MouseButtonEventArgs e)
   at System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
   at System.Windows.RoutedEventHandlerInfo.InvokeHandler(Object target, RoutedEventArgs routedEventArgs)
   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   at System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
   at System.Windows.UIElement.RaiseTrustedEvent(RoutedEventArgs args)
   at System.Windows.Input.InputManager.ProcessStagingArea()
   at System.Windows.Input.InputProviderSite.ReportInput(InputReport inputReport)
   at System.Windows.Interop.HwndMouseInputProvider.ReportInput(IntPtr hwnd, InputMode mode, Int32 timestamp, RawMouseActions actions, Int32 x, Int32 y, Int32 wheel)
   at System.Windows.Interop.HwndMouseInputProvider.FilterMessage(IntPtr hwnd, WindowMessage msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at System.Windows.Interop.HwndSource.InputFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
   at MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)
   at System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)
   at System.Windows.Application.RunDispatcher(Object ignore)
   at System.Windows.Application.RunInternal(Window window)
   at TEditXna.App.Main() in c:\projects\terraria-map-editor\TEditXna\obj\Release\App.g.cs:line 0 

**ETA:** I couldn't load the world in-game, so maybe it's just corrupted. I don't suppose you have any idea how to fix that?

**ETA:** I tried opening a previous version saved from before the crash, but it didn't work either. The backup created automatically by the game itself still works though. It's just several days old...

**Comment by enigmaquip** [comment]
It's corrupted somewhere in the middle of the file, in the tile area, I'm guessing if you looked at the file in a hex editor you'd see it just turn to zeros at the later half of the file..... Not much to be done as a corruption in tile data means everything after it is gone.... chests, signs, etc

**Comment by Samtarkras** [comment]
Same thing happens to me when i save sometimes no idea why hope this is fixed soon

**Comment by BinaryConstruct** [comment]
It allows you to force load a corrupt file, because on rare occasion you can recover some data.

**Comment by Samtarkras** [comment]
When it does give me the option it seems to load until it has to calculate colors, but i am not so sure it loaded cause it doesn't even show the npc faces like usual before it finishes calculating color

**Comment by Mewtwoy** [comment]
I just lost my server world because of this, backups did not work
somebody needs to fix this!

**Comment by missyTrinity** [comment]
I closed the ticket I opened because I have the SAME bug that I see here... Noe, I can't use my world and it works fine until this afternoon. I update my world and I hope I will recover it because I modified a lot of things... The "backup" did not work... I used the last version. Some idea why I can't load?

Thanks
![tedit bug](https://cloud.githubusercontent.com/assets/20374331/16782726/dfa6dbba-484e-11e6-9330-e5489141db8e.png)
